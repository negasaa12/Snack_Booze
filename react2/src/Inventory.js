

const alcoholicDrinks = [
  { "name": "Whiskey", "price": 40, "description": "A classic and strong distilled spirit made from various grains." },
  { "name": "Vodka", "price": 25, "description": "A clear and versatile spirit distilled from fermented grains or potatoes." },
  { "name": "Rum", "price": 30, "description": "A flavorful spirit made from sugarcane byproducts, often associated with tropical cocktails." },
  { "name": "Gin", "price": 35, "description": "A juniper-flavored spirit, perfect for martinis and gin and tonics." },
  { "name": "Tequila", "price": 45, "description": "A Mexican spirit made from the blue agave plant, known for its distinctive taste." },
  { "name": "Wine", "price": 20, "description": "An alcoholic beverage made from fermented grapes or other fruits." },
  { "name": "Beer", "price": 5, "description": "A refreshing alcoholic beverage brewed from grains, hops, and yeast." },
  { "name": "Cocktail", "price": 15, "description": "A mixed drink combining various spirits and flavors for a unique taste experience." }
];

const foodItems = [
  { "name": "Burger", "price": 5.99, "description": "A delicious sandwich made with a beef or plant-based patty, topped with various ingredients." },
  { "name": "Pizza", "price": 6.67, "description": "A classic Italian dish consisting of a flat, round crust topped with sauce, cheese, and various toppings." },
  { "name": "Pasta", "price": 7.45, "description": "A versatile dish made from unleavened dough, often served with various sauces." },
  { "name": "Salad", "price": 4.99, "description": "A healthy dish consisting of mixed greens, vegetables, and often topped with dressing." },
  { "name": "Steak", "price": 12.99, "description": "A high-quality cut of meat, typically beef, cooked to perfection." },
  { "name": "Sushi", "price": 10.50, "description": "A Japanese delicacy made with vinegared rice, fish, and vegetables." },
  { "name": "Sandwich", "price": 4.75, "description": "A convenient and delicious meal served between two slices of bread." },
  { "name": "Fries", "price": 3.99, "description": "Deep-fried potato strips, a popular side dish." }
];

  export {alcoholicDrinks, foodItems};