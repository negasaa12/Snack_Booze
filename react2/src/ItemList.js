import React from "react";



const ItemList = ({foodItems, alcoholicDrinks})=>{



    return(
        <>
        <h2> Snack & Booze </h2>
        
        <div>
         We have {foodItems.length} Amazing Snacks to choose from
        </div>
        <div>
            We have {alcoholicDrinks.length} Amazing Drinks to choose from
        </div>
        </>
    )
}

export default ItemList;