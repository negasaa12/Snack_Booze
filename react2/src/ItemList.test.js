import React from "react";
import {render, fireEvent} from "@testing-library/react";
import ItemList from "./ItemList";
import { foodItems, alcoholicDrinks } from "./Inventory";

const items = {
    snacks: foodItems,
    drinks: alcoholicDrinks
  };

 it("renders ItemList", ()=>{
    
      render(<ItemList items={items}/>)

 })

 

it("matches snapshot", ()=>{

    
    const { asFragment } = render(<ItemList items={items}/>);
    expect(asFragment()).toMatchSnapshot();
})
