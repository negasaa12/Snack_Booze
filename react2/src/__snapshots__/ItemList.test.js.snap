// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="itemList-container"
  >
    <h3
      class="itemList-h3"
    >
      Welcome to Snacks & Booze
    </h3>
    <p
      class="itemList-p"
    >
      We offer a wide variety of 16 delicious treats, including:
    </p>
    <ul
      class="itemList-ul"
    >
      <li
        class="itemList-li"
      >
        <strong>
          8 Kinds of Snacks
        </strong>
      </li>
      <li>
        <strong>
          8 Refreshing Drinks
        </strong>
      </li>
    </ul>
    <p
      class="itemList-p"
    >
      Whether you're craving something savory or looking for a refreshing beverage, we have something for everyone! Come and explore our delectable selection today.
    </p>
  </div>
</DocumentFragment>
`;
